### PR TITLE
Butunz

### PR DESCRIPTION
**Pull request by Arrinao**
Starting to work on the buttons. Added button images. These are supposed to have a highlightable text inside and they'll be pressable eventually

### CODE REVIEW

**Comment by Arrinao** [comment]
The letters inside buttons are still not centerized :/

File: tetris.py [hunk 1]
```diff
@@ -25,16 +28,56 @@
...
+
+highscores = tkinter.PhotoImage(file=(image_dir / "highscores.png"))
+highscores_hover = tkinter.PhotoImage(file=(image_dir / "hhighscores.png"))
+
+button_image1.tk.call(button_image1, 'copy', start, '-compositingrule', 'overlay')
+button_image2.tk.call(button_image2, 'copy', start_hover, '-compositingrule', 'overlay')
+button_image3.tk.call(button_image3, 'copy', gamemode, '-compositingrule', 'overlay')
+button_image4.tk.call(button_image4, 'copy', gamemode_hover, '-compositingrule', 'overlay')
+button_image5.tk.call(button_image5, 'copy', highscores, '-compositingrule', 'overlay')
+button_image6.tk.call(button_image6, 'copy', highscores_hover, '-compositingrule', 'overlay')
```
Comment (Akuli) [inline tetris.py:60]: Please use loops instead of copy/pasta.

File: tetris.py [hunk 2]
```diff
@@ -74,14 +117,14 @@ def run_gui():
     sidebar = tkinter.Frame(root, bg=D_GREY)
     sidebar.grid(row=1, column=1, sticky="nsw")
 
-    new_game_button = tkinter.Button(sidebar, text="start")
-    new_game_button.grid(sticky="n")
+    new_game_button_image = tkinter.Button(sidebar, image=button_image1, borderwidth=0, highlightthickness=0)
+    new_game_button_image.grid(sticky="n")
```
Comment (Akuli) [inline tetris.py:121]: wtf?

**Review by Akuli — CHANGES_REQUESTED** [review]

File: tetris.py [hunk 3]
```diff
@@ -25,16 +28,32 @@
 
 GameStatus = Enum("GameStatus", "in_progress, game_lost, paused")
 
+try:
+    image_dir = pathlib.Path(sys._MEIPASS)
+except AttributeError:
+    image_dir = pathlib.Path(__file__).parent / "images"
```
Comment (Akuli) [inline tetris.py:34]: The latest minesweeper has comments explaining what this code does. Please add them here too. They are needed. I explained the code to ThePhilgrim at least 2 times before we added the comments.

File: tetris.py [hunk 4]
```diff
@@ -25,16 +28,32 @@
...
+
+root = tkinter.Tk()
+root.resizable(False, False)
+
+button_images = {}
+for filename in ["start.png", "hstart.png", 'gamemode.png', 'hgamemode.png', 'highscores.png', 'hhighscores.png']:
+    transparent_image = tkinter.PhotoImage(file=(image_dir / filename))
+    button_images[filename] = tkinter.PhotoImage(file=image_dir / 'button.png')
+    button_images[filename].tk.call(button_images[filename], 'copy', transparent_image, '-compositingrule', 'overlay')
```
Comment (Akuli) [inline tetris.py:43]: All this code should go inside `run_gui` function.

File: tetris.py [hunk 5]
```diff
@@ -25,16 +28,32 @@
...
 
-def run_gui():
 
-    root = tkinter.Tk()
-    root.resizable(False, False)
+def highlight_icon_text(button_image, event):
+    return event.widget.config(image=button_image)
```
Comment (Akuli) [inline tetris.py:47]:
```suggestion
    event.widget.config(image=button_image)
```

File: tetris.py [hunk 6]
```diff
@@ -25,16 +28,32 @@
...
+for filename in ["start.png", "hstart.png", 'gamemode.png', 'hgamemode.png', 'highscores.png', 'hhighscores.png']:
+    transparent_image = tkinter.PhotoImage(file=(image_dir / filename))
+    button_images[filename] = tkinter.PhotoImage(file=image_dir / 'button.png')
+    button_images[filename].tk.call(button_images[filename], 'copy', transparent_image, '-compositingrule', 'overlay')
 
-def run_gui():
 
-    root = tkinter.Tk()
-    root.resizable(False, False)
+def highlight_icon_text(button_image, event):
```
Comment (Akuli) [inline tetris.py:46]: Function name is confusing. How about `set_button_image`?

**Review by Akuli — CHANGES_REQUESTED** [review]

File: tetris.py [hunk 7]
```diff
@@ -74,14 +88,21 @@ def run_gui():
     sidebar = tkinter.Frame(root, bg=D_GREY)
     sidebar.grid(row=1, column=1, sticky="nsw")
 
-    new_game_button = tkinter.Button(sidebar, text="start")
-    new_game_button.grid(sticky="n")
+    #image source https://cooltext.com/
```
Comment (Akuli) [inline tetris.py:91]: I love this comment. It's as short as it could possibly be but it contains all the information needed. Beautiful!

Comment (Arrinao) [reply tetris.py:96]: ♥

**Review by Akuli — CHANGES_REQUESTED** [review]
Delete button3.png

**Review by Akuli — APPROVED** [review]

**Pull request merged**